src/commonTest/kotlin/it/krzeminski/snakeyaml/engine/kmp/usecases/recursive/RecursiveSetTest.kt

```diff
@@ -0,0 +1,30 @@
+package it.krzeminski.snakeyaml.engine.kmp.usecases.recursive
+
+import io.kotest.assertions.throwables.shouldThrow
+import io.kotest.core.spec.style.FunSpec
+import io.kotest.matchers.shouldBe
+import it.krzeminski.snakeyaml.engine.kmp.api.Load
+import it.krzeminski.snakeyaml.engine.kmp.api.LoadSettings
+import it.krzeminski.snakeyaml.engine.kmp.exceptions.YamlEngineException
+import it.krzeminski.snakeyaml.engine.kmp.stringFromResources
+
+class RecursiveSetTest: FunSpec({
+    test("Fail to load map with recursive keys") {
+        val recursiveInput = stringFromResources("/recursive/recursive-set-1.yaml")
+        val load = Load()
+        shouldThrow< YamlEngineException> {
```

Comment:
formatting(non-blocking): looks like there is an extra space
```suggestion
        shouldThrow<YamlEngineException> {
```